### PR TITLE
Paging Helper Functions

### PR DESCRIPTION
**Pull request by tymefighter**

### CODE REVIEW

File: kernel/paging.c [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+*/
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress-of-packed-member"
+
+// Get offset into a page, it is valid for both physical and virtual addresses
+// Get least 12 bits
+uint64 getOffset(uint64 addr) {
+    return addr & 0x000000000000ffff;
+}
```
Comment (sahul343) [inline kernel/paging.c:40]: it's 12 or 16 bits. In comments, it's 12 bit but you are extracting 16 bits.

File: kernel/paging.c [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    -------------------------------------------------------
+*/
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress-of-packed-member"
+
+// Get offset into a page, it is valid for both physical and virtual addresses
+// Get least 12 bits
+uint64 getOffset(uint64 addr) {
+    return addr & 0x0000000000000fff;
```
Comment (sahul343) [inline kernel/paging.c:39]: resolved

File: kernel/paging.c (hunk 1 above)
Comment (tymefighter) [reply kernel/paging.c]: Oh it is 12, that is a very good catch !!
I'll correct it thanks !!

File: kernel/paging.c (hunk 2 above)
Comment (tymefighter) [reply kernel/paging.c:39]: Great !

**Comment by tymefighter** [comment]
> The branch I have opened for process and this one has some common implementations, merge conflicts can happen, so check that branch thoroughly

Will do it....

**Pull request merged**